### ISSUE TITLE
Next.js SSR monorepo: No ssrResources.json file

### ISSUE THREAD

**Issue by shishkin** [aws-amplify/amplify-console#2449]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-console/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-console/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-console/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Amplify Console feature

Monorepo

### Describe the bug

I'm deploying a next.js app that resides in a folder `packages/myapp` of my repo. I've figured out hot to setup the build settings for `next` to compile the app and the build step succeeds. However, the deploy step fails with the following:

```
2021-12-13T18:06:34 [INFO]: Beginning deployment for application d1vauylx53grjf, branch:main, buildId 0000000005
2021-12-13T18:06:34 [INFO]: Cannot find any generated SSR resources to deploy. If you intend for your app to be SSR, please check your app Service Role permissions. Otherwise, please check out our docs on how to setup your app to be detected as SSG (https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html#deploy-nextjs-app)
2021-12-13T18:06:34 [ERROR]: {"code":"7","message":"No ssrResources.json file"}
``` 

Where is the missing `ssrResources.json` file suppose to be. Next.js doesn't seem to generate it.

### Expected behavior

Amplify Console detects the app to be Next.js SSR and the build succeeds, so I expect the deploy step to work.

### Reproduction steps

1. Create app via Console

### Build Settings

```yaml
version: 1
applications:
  - appRoot: packages/myapp
    frontend:
      phases:
        preBuild:
          commands:
            - '(cd ../.. && yarn install --immutable)'
        build:
          commands:
            - '(cd ../.. && yarn run build)'
            - pwd
      artifacts:
        baseDirectory: .next
        files:
          - '**/*'
```

### Additional information

I'm using yarn. I've also tried to override the version of `next` to `12.0.7` which I'm using locally, but it had no effect.

**Comment by shishkin** [comment]
Not sure why you ask for App ID and region. This platform lacks confidentiality for sharing private customer details.

**Comment by Rafcin** [comment]
@shishkin You need to provide the App ID and region so the internal team can check out your logs and see what went wrong. You can always PM whoever picks up the case on Discord or send the information to the Amplify email. 

I've run into this issue before, are you using NX by chance or is this a standard mono repo?

**Comment by shishkin** [comment]
@Rafcin thanks for clarifications. Which discord is that?

No, I don't use NX. It's just a plain yarn workspaces structure with a next SSR app.

**Comment by Rafcin** [comment]
https://discord.gg/pt46FWbzqe

**Comment by hloriana** [comment]
Hi @shishkin thanks for raising this issue. I'm currently looking into this to see what could be causing this deployment error. I will update once I have more info.

**Comment by hloriana** [comment]
@shishkin can you go into your app settings in the Amplify console and ensure that there is a service role attached with permissions for backend deployment? Even though your app builds successfully, it appears that the service role is missing which is prohibiting proper SSR deployment.

**Comment by shishkin** [comment]
Thanks @hloriana. It is probably a permission issue. I just find Amplify so unintuitive and magical, that I gave up troubleshooting it. First, wasted so much time on CLI issues. Then thought I could use CDK construct to still use the SSR hosting. But eventually switched to `@sls-next/cdk-construct`. That's still experimental and has issues, but at least it's all OSS and straightforward to debug. Too bad Amplify can't offer an easy and reliable service.

**Comment by simaspoldantb** [comment]
I got same problem. with message
`[INFO]: Cannot find any generated SSR resources to deploy. If you intend for your app to be SSR, please check your app Service Role permissions. Otherwise, please check out our docs on how to setup your app to be detected as SSG (https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html#deploy-nextjs-app)`

`{"code":"7","message":"No ssrResources.json file"}`

this is case is wasting time, I have try out 2 days to solve this case, but still failed to Deploy nextjs.

there any guide to set role ?

**Comment by seokjungms** [comment]
> I got same problem. with message `[INFO]: Cannot find any generated SSR resources to deploy. If you intend for your app to be SSR, please check your app Service Role permissions. Otherwise, please check out our docs on how to setup your app to be detected as SSG (https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html#deploy-nextjs-app)`
> 
> `{"code":"7","message":"No ssrResources.json file"}`
> 
> this is case is wasting time, I have try out 2 days to solve this case, but still failed to Deploy nextjs.
> 
> there any guide to set role ?

According to the aws amplify nextjs doc, amplify detects whether it's SSG by seeing the 1) package.json has "next-export" script and 2) Build settings have "out" as baseDirectory. In my case, I was missing next-export in my pakcage.json when I first created Amplify app. Even if I changed the package.json and build settings, the error still occured. I also downgraded the next.js according to https://github.com/aws-amplify/amplify-hosting/issues/2641, but had the same error.

I deleted the app and created a new one. This solved the error. I think the Amplify detects SSG or SSR when the app is first created by looking at the package.json. While I was creating the app, the console indicated that my app was SSG. I don't remember whether the first app, which had the error, indicated SSR when I created because it's already been deleted. 

Probably the Amplify detected my app as SSR in my first launching and the configuration was set accordingly. I got the hint from this page https://github.com/aws-amplify/amplify-hosting/blob/main/FAQ.md#ssr. Since I was only starting, I decided to launch a new one instead of converting it to SSG.

Hope this helps you and anyone googling this error.
